File: garden_advice.js
// GApp - Gardening Advice Application


function getUserInput() {
    // Ask the user to enter the current season.
    let season = prompt(
        "Enter the season (summer, winter, spring, autumn):"
    );

    // Ask the user to enter the plant type.
    let plantType = prompt(
        "Enter the plant type (flower or vegetable):"
    );

    // Remove extra spaces and convert input to lowercase.
    season = season.trim().toLowerCase();
    plantType = plantType.trim().toLowerCase();

    return {
        season: season,
        plantType: plantType
    };
}


function getSeasonAdvice(season) {
    // Store advice for each season in an object.
    const seasonAdvice = {
        summer: "Water your plants regularly and provide some shade.",
        winter: "Protect your plants from frost with covers.",
        spring: "Prepare the soil and start planting new plants.",
        autumn: "Remove dead leaves and prepare plants for colder weather."
    };

    // Return the advice if the season exists.
    if (seasonAdvice[season]) {
        return seasonAdvice[season];
    }

    return "No advice for this season.";
}


function getPlantAdvice(plantType) {
    // Store advice for different plant types in an object.
    const plantAdvice = {
        flower: "Use fertiliser to encourage blooms.",
        vegetable: "Keep an eye out for pests!"
    };

    // Return advice if the plant type exists.
    if (plantAdvice[plantType]) {
        return plantAdvice[plantType];
    }

    return "No advice for this type of plant.";
}


function recommendPlants(season) {
    // Store recommended plants for each season.
    const plantRecommendations = {
        summer: [
            "Sunflowers",
            "Tomatoes",
            "Peppers"
        ],

        winter: [
            "Spinach",
            "Broccoli",
            "Cabbage"
        ],

        spring: [
            "Petunias",
            "Carrots",
            "Lettuce"
        ],

        autumn: [
            "Pansies",
            "Onions",
            "Beetroot"
        ]
    };

    // Return the plant recommendations for the season.
    if (plantRecommendations[season]) {
        return plantRecommendations[season];
    }

    return [];
}


function displayAdvice(season, plantType) {
    // Get advice based on the selected season.
    const seasonAdvice = getSeasonAdvice(season);

    // Get advice based on the selected plant type.
    const plantAdvice = getPlantAdvice(plantType);

    // Get recommended plants for the season.
    const recommendedPlants = recommendPlants(season);

    // Combine the advice.
    const advice =
        seasonAdvice + "\n" +
        plantAdvice;

    console.log("--- Gardening Advice ---");
    console.log(advice);

    // Display recommended plants.
    if (recommendedPlants.length > 0) {
        console.log(
            "\nRecommended plants for " + season + ":"
        );

        for (let plant of recommendedPlants) {
            console.log("- " + plant);
        }
    } else {
        console.log(
            "\nNo plant recommendations available for this season."
        );
    }
}


function main() {
    // Get the user's input.
    const userInput = getUserInput();

    // Display the gardening advice.
    displayAdvice(
        userInput.season,
        userInput.plantType
    );
}


// Run the application.
main();

File: garden_advice.py
# GApp - Gardening Advice Application


def get_user_input():
    """
    Ask the user to enter the current season and plant type.
    The input is converted to lowercase to make comparisons easier.
    """
    season = input(
        "Enter the season (summer, winter, spring, autumn): "
    ).strip().lower()

    plant_type = input(
        "Enter the plant type (flower or vegetable): "
    ).strip().lower()

    return season, plant_type


def get_season_advice(season):
    """
    Return gardening advice based on the selected season.
    """

    season_advice = {
        "summer": (
            "Water your plants regularly and provide some shade."
        ),
        "winter": (
            "Protect your plants from frost with covers."
        ),
        "spring": (
            "Prepare the soil and start planting new plants."
        ),
        "autumn": (
            "Remove dead leaves and prepare plants for colder weather."
        ),
    }

    return season_advice.get(
        season,
        "No advice for this season."
    )


def get_plant_advice(plant_type):
    """
    Return gardening advice based on the selected plant type.
    """

    plant_advice = {
        "flower": (
            "Use fertiliser to encourage blooms."
        ),
        "vegetable": (
            "Keep an eye out for pests!"
        ),
    }

    return plant_advice.get(
        plant_type,
        "No advice for this type of plant."
    )


def recommend_plants(season):
    """
    Recommend plants that are suitable for the selected season.
    """

    plant_recommendations = {
        "summer": [
            "Sunflowers",
            "Tomatoes",
            "Peppers",
        ],
        "winter": [
            "Spinach",
            "Broccoli",
            "Cabbage",
        ],
        "spring": [
            "Petunias",
            "Carrots",
            "Lettuce",
        ],
        "autumn": [
            "Pansies",
            "Onions",
            "Beetroot",
        ],
    }

    return plant_recommendations.get(
        season,
        []
    )


def display_advice(season, plant_type):
    """
    Generate and display gardening advice and plant recommendations.
    """

    # Get advice based on the entered season.
    season_advice = get_season_advice(season)

    # Get advice based on the entered plant type.
    plant_advice = get_plant_advice(plant_type)

    # Get recommended plants for the season.
    recommended_plants = recommend_plants(season)

    # Combine the gardening advice.
    advice = season_advice + "\n" + plant_advice

    print("\n--- Gardening Advice ---")
    print(advice)

    # Display plant recommendations if the season is recognised.
    if recommended_plants:
        print("\nRecommended plants for", season + ":")

        for plant in recommended_plants:
            print("-", plant)
    else:
        print(
            "\nNo plant recommendations available "
            "for this season."
        )


def main():
    """
    Main function used to run the gardening application.
    """

    # Ask the user for the season and plant type.
    season, plant_type = get_user_input()

    # Display gardening advice.
    display_advice(season, plant_type)


# Run the program.
# if __name__ == "__main__":
main()
